WooCommerce/Classes/Tools/URLOpener.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+import UIKit
+
+protocol URLOpener {
+    func open(_ url: URL)
+}
+
+/// Uses the UIApplication API to open the url
+///
+struct ApplicationURLOpener: URLOpener {
+    func open(_ url: URL) {
+        UIApplication.shared.open(url, options: [:], completionHandler: nil)
```

Comment:
Should we check with `canOpenURL()` first? or is not needed?

Reply:
Good point! I considered that, and I think at this point it is not necessary:

- There is no risk in calling `open(_:options:completionHandler:)` directly. If the app cannot handle the URL schema, it will return success false in the completion handler, a no-op for us.
- In our case, we do not need to react if the URL cannot be opened.
- Since the universal links we get have the schema `https` and that can always be handled by the application, it is sure that the app can open the url.